### ISSUE TITLE
Wiki editing access ...

### ISSUE THREAD

**Issue by hholzgra** [mapnik/mapnik#4181]
What would it take to get wiki editing access?

**Comment by artemp** [comment]
@hholzgra - Assuming you're offering to improve and/or update wiki, sounds great and welcome! Github doesn't provide an easy way to manage wiki access unfortunately. Could you demo what changes you're proposing so we can figure our appropriate action, please. I think it's possible to clone Wiki repo though I haven't tried myself. 
/cc @springmeyer

**Comment by hholzgra** [comment]
>  Assuming you're offering to improve and/or update wiki

that's the plan, more or less ...

>  Could you demo what changes you're proposing so we can figure our appropriate action, please

I've started to collect some missing pieces from the source code and other sources, and roll my own "missing manual" for a little while already. I've been using AsciiDoc, not MarkDown, for that so far, but I think you get the idea. 

Repository for that is https://github.com/hholzgra/mapnik-lost-manual and processed result can be found on https://get-map.org/mapnik-lost-manual

>  I think it's possible to clone Wiki repo though I haven't tried myself.

It can be cloned locally, yes, but there doesn't seem to be support for things like pull requests with that :(

**Comment by hholzgra** [comment]
Some more background on me: I have been running the MapOSMatic instance on print.get-map.org (previously maposmatic.osm-baustelle.de) since somewhere in 2016, and am trying to keep as many osm2pgsql schema based styles running there ...

**Comment by artemp** [comment]
@hholzgra - It looks like there is a consensus to not use github wiki for documentation : https://github.com/mapnik/mapnik/issues/3543 and there are:

**Comment by hartmut-mariadb** [comment]
Hm, mapnik-reference is incomplete in itself, it is OK for symbolizers, mostly, but e.g. completely missing out on data sources. mapnik/docs mentioned in #3543 never really went anywhere, same for the sphinx-docs repository. I never really looked into node-mapnik, but that doesn't look too complete either ...

So for now I'll just continue with what I started already ...

**Comment by artemp** [comment]
You're right there are lots of missing bits and it's hard to find/navigate etc etc. Just for completeness here's link to published `node-mapnik`  docs in case you didn't see it: node-mapnik docs - https://mapnik.org/documentation/node-mapnik/3.6/
I thought somewhat similar approach would work better than github wiki, cheers.

**Comment by mathisloge** [comment]
@artemp maybe we should start writing doxygen comments into the source code and dedicated documentation files which gets included in the tooling. 

https://doxygen.nl/ together with https://mcss.mosra.cz/documentation/doxygen/

The CI can then auto publish the whole documentation to a github page. I can start with a basic setup and some description e.g. of the cmake installation etc.

**Comment by hholzgra** [comment]
While Doxgen is great for API documentation I don't think I would use it for general documentation. That's why I tried to go for an AsciiDoc document for user level documentation instead. This could also be tightly coupled with the images generated by the test suite to visualize the effects of different Mapnik XML tags and their various attributes.

The original Github Wiki did already do a somewhat good job on showing how output could be generated using either XML or the C++ and Python APIs, but unfortunately it is far from complete on this, and different wiki pages reflect different Mapnik version states.

That's why I originally asked for Wiki edit access, but eventually ended up rolling my own "missing manual" project, but unfortunately 
I wasn't able to invest as much time into this as I had originally hoped for, and so https://get-map.org/mapnik-lost-manual/book.html / https://get-map.org/mapnik-lost-manual is basically still mostly an incomplete brain dump, and not a real alternative to the Wiki (yet)
